path/path.go

```diff
@@ -30,23 +30,17 @@ func GetPart(path string) (part string, nextPath string) {
 	return
 }
 
-// GetNameFromPart gets node name from path part
+// GetNameFromPart gets node name from path part, braces are not stripped so that we can check which node is associated with a parameter
 func GetNameFromPart(pathPart string) (name string, exp string) {
 	name = pathPart
 
-	if pathPart[0] == '{' {
-		name = pathPart[1 : len(pathPart)-1]
-
-		if parts := strings.Split(name, ":"); len(parts) == 2 {
-			name = parts[0]
-			exp = parts[1]
-		}
-
-		if name == "" {
-			panic("Empty wildcard name")
-		}
+	if parts := strings.Split(name, ":"); len(parts) == 2 {
+		name = parts[0]
+		exp = parts[1]
+	}
 
-		return
+	if name == "" {
```

Comment:
Because you moved it outside the inner if  now the case after split is not handled `name = parts[0]`

Reply:
Please revert

Reply:
Yeah now I get why the params were missing from context.Params after this change and also made other tests fail.
```
if exp != "" {
		static.maxParamsSize++
		node = withRegexp(static, regexp.MustCompile(exp))
	} else if name != pathPart {
		static.maxParamsSize++
		node = withWildcard(static)
	} else {
		node = static
	}
```